who-metrics-ui/src/components/RepositoriesTable.tsx

```diff
@@ -224,6 +225,26 @@ const defaultFilters: Filter = {
   },
 };
 
+// Helper for generating the csv blob
+const generateCSV = (data: Repo[]): Blob => {
+  const output = json2csv(data);
+  return new Blob([output], { type: 'text/csv' });
+};
+
+// Helper for downloading the csv
+// There are warnings for Opera Mini incompatibility, but only 0.07% of users use that browser
+const saveFile = async (contents: Blob) => {
```

Comment:
Just curious, what's the difference between creating the `a` element here on the fly vs. just creating it once and including it in the DOM from the get go? Do we need to create it on the fly so that we can dynamically generate the CSV blob?

Reply:
I imagine screen readers/things that read links on the page will see the a with contents and I'm not sure that's what we want. Maybe the best case here would be to dynamically create and destroy this element on each button click

Reply:
As long as we use the correct CSS classes it should be hidden visually and from screen readers. I haven't double checked but I think this is right: https://a11y-guidelines.orange.com/en/articles/accessible-hiding/#:~:text=The%20conventional%20way%20is%20to,by%20Assistive%20technologies%20(AT).

Reply:
Went with a dedicated package instead to increase compatibility and around these hacks